Mission7/schema.sql

```diff
@@ -0,0 +1,67 @@
+-- 사용자(users Table)
+CREATE TABLE users (
+  id SERIAL PRIMARY KEY,
+  email VARCHAR(255) NOT NULL UNIQUE,
+  nickname VARCHAR(100) NOT NULL,
+  password TEXT NOT NULL,
+  created_at TIMESTAMP DEFAULT NOW(),
+  updated_at TIMESTAMP DEFAULT NOW(),
+);
+
+-- 게시글(article Table)
+CREATE TABLE article (
+  id SERIAL PRIMARY KEY,
+  title VARCHAR(255) NOT NULL,
+  content TEXT NOT NULL,
+  user_id int,
+  created_at TIMESTAMP DEFAULT NOW(),
+  updated_at TIMESTAMP DEFAULT NOW(),
+  FOREING KEY (user_id) REFERENCES users(id) ON DELETE CASCADE
+);
+
+-- 제품(product)
+CREATE TABLE product (
+  id  SERIAL PRIMARY KEY,
+  name VARCHAR(255) NOT NULL,
+  description TEXT NOT NULL,
+  price INT NOT NULL,
+  tags TEXT[] NOT NULL,
+  user_id int,
+  created_at TIMESTAMP DEFAULT NOW(),
+  updated_at TIMESTAMP DEFAULT NOW(),
+  FOREING KEY (user_id) REFERENCES users(id) ON DELETE CASCADE
+)
+
+-- 댓글(commend)
+CREATE TABLE commend (
+  id SERIAL PRIMARY KEY,
+  content TEXT NOT NULL,
+  product_id int,
+  article_id int,
+  user_id int,
+  created_at TIMESTAMP DEFAULT NOW(),
+  updated_at TIMESTAMP DEFAULT NOW(),
+  FOREING KEY (user_id) REFERENCES users(id) ON DELETE CASCADE,
+  FOREING KEY (product_id) REFERENCES product(id) ON DELETE CASCADE,
+  FOREING KEY (article_id) REFERENCES article(id) ON DELETE CASCADE
+)
+
+-- 즐겨찾기(favorite)
+CREATE TABLE favorite (
+  id SERIAL PRIMARY KEY,
+  product_id int UNIQUE,
+  user_id int UNIQUE,
+  FOREING KEY (product_id) REFERENCES product(id) ON DELETE CASCADE,
+  FOREING KEY (user_id) REFERENCES users(id) ON DELETE CASCADE
+)
+
+-- 좋아요 (like)
+CREATE TABLE like (
+  id SERIAL PRIMARY KEY,
+  article_id int UNIQUE,
+  user_id int UNIQUE,
+  product_id int UNIQUE,
+  FOREING KEY (product_id) REFERENCES product(id) ON DELETE CASCADE,
+  FOREING KEY (article_id) REFERENCES article(id) ON DELETE CASCADE,
+  FOREING KEY (user_id) REFERENCES users(id) ON DELETE CASCADE
+)
```

Comment:
유저, 아티클, 프로덕트는 여러개 있을 수 있죠. 이런 경우에는 복합 Primary Key를 만들어야합니다.
```suggestion
CREATE TABLE user_liked_articles (
  user_id INTEGER NOT NULL REFERENCES users(id) ON DELETE CASCADE,
  article_id INTEGER NOT NULL REFERENCES articles(id) ON DELETE CASCADE,
  PRIMARY KEY (user_id, article_id)
);

```
위와 같이 만들 수 있고 like에도 identifier가 필요하다면 id를 추가해볼수도 있겠군요.(대부분의 케이스는 필요없을 듯 합니다.)